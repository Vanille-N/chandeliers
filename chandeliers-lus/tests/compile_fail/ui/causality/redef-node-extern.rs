//! Duplicate definition of an `extern node`. It
//! is not possible for chandeliers to catch this, so
//! here we are verifying that rustc properly catches it
//! and puts the right spans.
chandeliers_lus::decl! {
    extern node foo(i : int) returns (i : int);
}

chandeliers_lus::decl! {
    extern node bar(i : int; i : bool) returns ();
}

fn main() {}
