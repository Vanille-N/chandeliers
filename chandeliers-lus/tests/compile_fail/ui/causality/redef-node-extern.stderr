error: Attempt to redefine i, when a prior item already defines it
 --> tests/compile_fail/ui/causality/redef-node-extern.rs:6:39
  |
6 |     extern node foo(i : int) returns (i : int);
  |                                       ^
  |
note: Already defined here
 --> tests/compile_fail/ui/causality/redef-node-extern.rs:6:21
  |
6 |     extern node foo(i : int) returns (i : int);
  |                     ^

error: Attempt to redefine i, when a prior item already defines it
  --> tests/compile_fail/ui/causality/redef-node-extern.rs:10:30
   |
10 |     extern node bar(i : int; i : bool) returns ();
   |                              ^
   |
note: Already defined here
  --> tests/compile_fail/ui/causality/redef-node-extern.rs:10:21
   |
10 |     extern node bar(i : int; i : bool) returns ();
   |                     ^
